zc_events/backends/rabbitmqredis.py

```diff
@@ -0,0 +1,284 @@
+        logger.info('zc_events finished name={name} data={data} response={response}'.format(
+            name=name, data=data, response=response))
+        backend.respond(request.response_key, response)
+
+
+def _get_raw_response(redis_client, response_key):
+    try:
+        key_and_response = redis_client.blpop(response_key, 30)
+        if key_and_response is None:
+            raise RequestTimeout(detail='Timed out waiting for redis response')
+
+        response = key_and_response[1]  # Index [0] is the response_key
+        logger.debug('zc_events got response response_key={response_key} response={response}'.format(
+            response_key=response_key,
+            response=response
+        ))
+    except Exception as e:
+        msg = str(e)
+        ex_type = e.__class__.__name__
+        trace = traceback.format_exc()
+        logger.exception(
+            'zc_events exception waiting for response response_key={response_key} '
+            'exception={ex} message={msg} trace={trace}'.format(
+                response_key=response_key,
+                ex=ex_type,
+                msg=msg,
+                trace=trace
+            )
+        )
+        response = json.dumps(_format_exception_response(ex_type, msg, trace))
+    return response
+
+
+def _respond(redis_client, response_key, data):
+    result = redis_client.rpush(response_key, json.dumps(data))
+    redis_client.expire(response_key, 60)
+    return result
+
+
+def _place_on_queue(pika_pool, events_exchange, routing_key, priority, event_body):
+    event_queue_name = '{}-events'.format(settings.SERVICE_NAME)
+    queue_arguments = {
+        'x-max-priority': 10
+    }
+    response = None
+    logger.debug(
+        'zc_events placing on queue with the following '
+        'events_exchange={events_exchange} routing_key={routing_key} '
+        'event_body={event_body} priority={priority}'.format(
+           events_exchange=events_exchange, routing_key=routing_key, event_body=event_body, priority=priority
+        )
+    )
+    with pika_pool.acquire() as cxn:
+        cxn.channel.queue_declare(queue=event_queue_name, durable=True, arguments=queue_arguments)
+        response = cxn.channel.basic_publish(
+            events_exchange,
+            routing_key,
+            event_body,
+            pika.BasicProperties(
+                content_type='application/json',
+                content_encoding='utf-8',
+                priority=priority
+            )
+        )
+
+    if not response:
+        raise EmitEventException("Message may have failed to deliver")
+    return response
+
+
+def _format_data(data, method, key):
+    data['_backend'] = {
+        'type': 'rabbitmqfanout',
+        'method': method,
+        'key': key
+    }
+    return {
+        'task': 'microservice.event',
+        'id': data['id'],
+        'args': [key, data],
+        'response_key': data['response_key']
+    }
+
+
+class RabbitMqFanoutBackend:
```

Comment:
One last class that should inherit from `object`, but other than that it looks good!